scripts/dts/edtlib.py

```diff
@@ -169,10 +169,19 @@ def _init_compat2binding(self, bindings_dirs):
             "|".join(re.escape(compat) for compat in dt_compats)
         ).search
 
-        self._binding_paths = _binding_paths(bindings_dirs)
+        # Used by 'include:'
+        self._binding_paths = []
+        # Stores paths relative to the binding directory. Used to make make
+        # output deterministic.
+        relpaths = []
+
+        for bdir in bindings_dirs:
+            for bpath in _binding_paths(bdir):
+                self._binding_paths.append(bpath)
+                relpaths.append(os.path.relpath(bpath, bdir))
```

Comment:
This part of the code is confusing because there are not just two but three parts involved in the paths here:

1. ZEPHYR_BASE
2. `dts/bindings`
3. `some/dir/file.yaml`

I think I remember what this particular relpath() does but barely and that's after me having designed, written and thoroughly tested the first version of this in PR #19333! So imagine someone new to this code...

=> this deserves: - either more comments clarifying what is relative to what (some `dts/bindings` example wouldn't hurt); - or super-smart variable names, much more elaborated than these. Adding comments seems obviously faster and easier ("If I Had More Time, I Would Have Written a Shorter Letter - Blaise Pascal)

Reply:
Lengthened the variable names and added a comment. The `relpath()` is to remove the path to the binding directory, turning e.g. `path/to/zephyr/dts/bindings/sub/foo.yaml` into `sub/foo.yaml`.